v3/src/components/graph/adornments/adornment.tsx

```diff
@@ -13,22 +14,16 @@ import "./adornment.scss"
 interface IProps {
   adornment: IAdornmentModel
   cellKey: Record<string, string>
-  topCats: string[] | number[]
-  rightCats: string[] | number[]
 }
 
-export const Adornment = observer(function Adornment(
-  {adornment, cellKey, topCats, rightCats}: IProps
-) {
-  const graphModel = useGraphContentModelContext(),
-    layout = useGraphLayoutContext(),
-    subPlotWidth = topCats.length > 0
-                     ? layout.plotWidth / topCats.length
-                     : layout.plotWidth,
-    subPlotHeight = rightCats.length > 0
-                      ? layout.plotHeight / rightCats.length
-                      : layout.plotHeight
-
```

Comment:
I didn't get to the bottom of why, but when removing a top split, `topCats` was not changing. For example, if you were using Mammals and added Diet to the top split, `topCats` became `['meat', 'plants', 'both']`, but when you removed Diet from the top split, it would still be `['meat', 'plants', 'both']`. The same issue was true for `rightCats`, too. 

`topCats` and `rightCats` were being passed as props to `Adornment` by `Adornments` where they were set by calling `layout.getAxisMultiScale("top").categorySet` and `layout.getAxisMultiScale("rightSplit").categorySet`. The values weren't being properly updated there either.

It seems like the new method for setting `subPlotWidth` and `subPlotHeight` here is more concise anyway and follows a pattern already used elsewhere in CODAP v3. So I only mention the above in case it seems like there may be a larger problem with `useGraphLayoutContext`, although there may not be.

Reply:
I suspect it may be a caching/invalidation issue in `CategorySet` rather than a problem with `useGraphLayoutContext`. I'll add a PT story about it.